tests/integration/test_replicated_merge_tree_encryption_codec/test.py

```diff
@@ -91,7 +91,9 @@ def test_different_keys():
     copy_keys(node2, "key_b")
     create_table()
 
-    insert_data()
+    # Insert two blocks without duplicated blocks to force each replica to actually fetch parts from another replica.
```

Comment:
What's the difference with the current `insert_data()`? It's also inserting to both replicas:

```
    node1.query("INSERT INTO tbl VALUES (1, 'str1')")
    node2.query("INSERT INTO tbl VALUES (1, 'str1')")  # Test deduplication
    node2.query("INSERT INTO tbl VALUES (2, 'str2')")
```

Reply:
The difference is in that line:
```
node2.query("INSERT INTO tbl VALUES (1, 'str1')") # Test deduplication      (D)
```

When `insert_data()` was written here the following two scenarios were possible:

**Scenario 1.**
`node1` writes a part for `(1, 'str1')`, then
`node2` downloads this part from `node1`, then
this line (D) doesn't actually insert anything because of deduplication.

In this scenario after `SYSTEM SYNC REPLICA` what we have is that both `node1` and `node2` contain part `(1, 'str1')` encrypted with `key_a` and part `(2, 'str2')` encrypted with `key_b`. So `SELECT * FROM tbl` fails on both nodes because each replica contains only one encryption key.

**Scenario 2.**
`node1` writes a part for `(1, 'str1')`, then
`node2` writes a part for `(1, 'str1')` in this line (D), then
`node2` doesn't fetch `(1, 'str1')` from `node1` because of deduplication.

In this scenario after `SYSTEM SYNC REPLICA` what we have is that `node1` contains part `(1, 'str1')` encrypted with `key_a` and part `(2, 'str2')` encrypted with `key_b`, and `node2` contains parts `(1, 'str1')` and `(2, 'str2')` both encrypted with `key_b` . So `SELECT * FROM tbl` fails on `node1` but succeeds on `node2`, making this test flaky.

Having duplicated blocks is not necessary for this test, so I decided not to use them.

Reply:
I'll add a comment about that to the test.

Reply:
I see. Thanks for the explanation!